### PR TITLE
#7503 Update certificates

### PR DESCRIPTION
**Pull request by planetf1**
Signed-off-by: Nigel Jones <nigel.l.jones+git@gmail.com>

## Description

* Updates cert generation script to retain existing CAs by default

Using the same CAs as they are valid for 10 years, this also means no need to recopy trust stores which may be used in client apps etc

* Allows dup certificates for same subject name

Currently a new cert cannot be generated for the same subject. However this doesn't take into account validity, and we therefore need dups since we'll have a new certificate, and an old one that have been created

* Generate the new certificates

All client/server certs generated by this tool are updated, and the copies encoded within this tool have been done.

## Testing

Build verification

Checked all .pem & .p12 files are up to date - except for certification authorities:
```
egeria git:(issue7503) find . -name '*.p12' -exec ls -la {} \;
-rw-------@ 1 jonesn  staff  6250  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaReactUIServer.p12
-rw-r--r--@ 1 jonesn  staff  1927  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaRootCA.p12
-rw-------@ 1 jonesn  staff  3020  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaClient.p12
-rw-------@ 1 jonesn  staff  6226  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaUIChassis.p12
-rw-------@ 1 jonesn  staff  3050  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaReactUIClient.p12
-rw-------@ 1 jonesn  staff  6250  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaServerChassis.p12
-rwxr-xr-x  1 jonesn  staff  1927  9 Mar 12:36 ./open-metadata-distribution/open-metadata-assemblies/build/unpacked/egeria-4.0-SNAPSHOT-distribution.tar.gz/truststore.p12
-rwxr-xr-x  1 jonesn  staff  6250  9 Mar 12:36 ./open-metadata-distribution/open-metadata-assemblies/build/unpacked/egeria-4.0-SNAPSHOT-distribution.tar.gz/keystore.p12
-rw-r--r--@ 1 jonesn  staff  1927  9 Mar 12:30 ./truststore.p12
-rw-r--r--@ 1 jonesn  staff  3020  9 Mar 12:30 ./EgeriaClient.p12
-rw-r--r--  1 jonesn  staff  1927  9 Mar 12:35 ./open-metadata-implementation/user-interfaces/ui-chassis/ui-chassis-spring/build/resources/main/truststore.p12
-rw-r--r--  1 jonesn  staff  6226  9 Mar 12:35 ./open-metadata-implementation/user-interfaces/ui-chassis/ui-chassis-spring/build/resources/main/keystore.p12
-rw-r--r--@ 1 jonesn  staff  1927  9 Mar 12:30 ./open-metadata-implementation/user-interfaces/ui-chassis/ui-chassis-spring/src/main/resources/truststore.p12
-rw-r--r--@ 1 jonesn  staff  6226  9 Mar 12:30 ./open-metadata-implementation/user-interfaces/ui-chassis/ui-chassis-spring/src/main/resources/keystore.p12
-rw-r--r--  1 jonesn  staff  1927  9 Mar 12:35 ./open-metadata-implementation/server-chassis/server-chassis-spring/build/resources/main/truststore.p12
-rw-r--r--  1 jonesn  staff  6250  9 Mar 12:35 ./open-metadata-implementation/server-chassis/server-chassis-spring/build/resources/main/keystore.p12
-rw-r--r--@ 1 jonesn  staff  1927  9 Mar 12:30 ./open-metadata-implementation/server-chassis/server-chassis-spring/src/main/resources/truststore.p12
-rw-r--r--@ 1 jonesn  staff  6250  9 Mar 12:30 ./open-metadata-implementation/server-chassis/server-chassis-spring/src/main/resources/keystore.p12
➜  egeria git:(issue7503) find . -name '*.pem' -exec ls -la {} \;
-rw-------@ 1 jonesn  staff  1874  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaUIChassis.key.pem
-rw-r--r--@ 1 jonesn  staff  1659  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaClient.cert.pem
-rw-r--r--@ 1 jonesn  staff  4163  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/tmp.pem
-rw-r--r--@ 1 jonesn  staff  956  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaReactUIClient.csr.pem
-rw-r--r--@ 1 jonesn  staff  956  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaServerChassis.csr.pem
-rw-r--r--@ 1 jonesn  staff  1830  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaServerChassis.cert.pem
-rw-r--r--@ 1 jonesn  staff  1830  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaReactUIServer.cert.pem
-rw-r--r--@ 1 jonesn  staff  948  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaClient.csr.pem
-rw-r--r--@ 1 jonesn  staff  956  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaReactUIServer.csr.pem
-rw-r--r--@ 1 jonesn  staff  1822  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaUIChassis.cert.pem
-rw-------@ 1 jonesn  staff  1874  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaClient.key.pem
-rw-r--r--@ 1 jonesn  staff  1996  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaRootCA/newcerts/1000.pem
-rw-r--r--@ 1 jonesn  staff  2167  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaRootCA/certs/EgeriaRootCA.cert.pem
-rw-r--r--@ 1 jonesn  staff  3326  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaRootCA/private/EgeriaRootCA.key.pem
-rw-------@ 1 jonesn  staff  1874  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaReactUIServer.key.pem
-rw-------@ 1 jonesn  staff  1874  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaReactUIClient.key.pem
-rw-------@ 1 jonesn  staff  1874  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaServerChassis.key.pem
-rw-r--r--@ 1 jonesn  staff  1683  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/csr/EgeriaIntermediateCA.csr.pem
-rw-r--r--@ 1 jonesn  staff  1667  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1008.pem
-rw-r--r--@ 1 jonesn  staff  1659  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1009.pem
-rw-r--r--@ 1 jonesn  staff  1822  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1001.pem
-rw-r--r--@ 1 jonesn  staff  1830  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1000.pem
-rw-r--r--@ 1 jonesn  staff  1830  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1002.pem
-rw-r--r--@ 1 jonesn  staff  1667  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1003.pem
-rw-r--r--@ 1 jonesn  staff  1830  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1007.pem
-rw-r--r--@ 1 jonesn  staff  1822  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1006.pem
-rw-r--r--@ 1 jonesn  staff  1659  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1004.pem
-rw-r--r--@ 1 jonesn  staff  1830  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/newcerts/1005.pem
-rw-r--r--@ 1 jonesn  staff  1996  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/certs/EgeriaIntermediateCA.cert.pem
-rw-r--r--@ 1 jonesn  staff  3326  7 Mar 12:00 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaIntermediateCA/private/EgeriaIntermediateCA.key.pem
-rw-r--r--@ 1 jonesn  staff  1667  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaReactUIClient.cert.pem
-rw-r--r--@ 1 jonesn  staff  952  9 Mar 12:30 ./open-metadata-resources/open-metadata-deployment/certificates/EgeriaUIChassis.csr.pem
```

## Additional notes

Certs used outside the core Egeria repo will need updating by their owners.
This likely means UI components @davidradl @sarbull

### CODE REVIEW

**Comment by planetf1** [comment]
@lpalashevski @mandy-chessell As per our discussion earlier, this PR updates our self signed certs as part of #7503
If changes look ok we can merge

**Review by lpalashevski — APPROVED** [review]

File: open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh [hunk 1]
```diff
@@ -43,112 +43,114 @@ CA_CHAIN=EgeriaCAChain
...
-printf "\n\n---- Create truststore (rootCA) ----\n\n"
-
-# -- will use keytool instead, can't seem to get a cert exported via openssl ...
-#openssl pkcs12 -export -in ${rootCA}/certs/${rootCA}.cert.pem -out ${rootCA}.p12 \
-#          -passin pass:${KEYPASS} -passout pass:${KEYPASS} -nokeys -cacerts
-keytool -importcert -noprompt -keystore ${rootCA}.p12 -storepass ${STOREPASS} -alias ${rootCA} \
-        -file ${rootCA}/certs/${rootCA}.cert.pem
-
+if [[ ! -z $CA_CLEAN ]]; then
+  rm -f *.pem
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh]: fixed

File: open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh [hunk 2]
```diff
@@ -43,112 +43,114 @@ CA_CHAIN=EgeriaCAChain
...
+  rm -f *.p12
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh]: fixed

File: open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh [hunk 3]
```diff
@@ -43,112 +43,114 @@ CA_CHAIN=EgeriaCAChain
...
+  # ---
+  # Root Certificate Authority
+  #
+  # Generates a root Certificate Authority for all the certs we create here in Egeria
+  # ---
+  # CA/CA_POLICY are variables used in the openssl.cnf file -- we need to use two different contexts -
+  # for both the root cert authority, and then the intermediate cert authority
+  export CA=${rootCA}
+  export CA_POLICY=policy_strict
+  printf "\n\n---- Generating root CA ($rootCA)  ----\n\n"
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh]: fixed

File: open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh [hunk 4]
```diff
@@ -43,112 +43,114 @@ CA_CHAIN=EgeriaCAChain
...
+
+  # ---
+  # Intermediate Certificate Authority
+  #
+  # This is an intermediate Certificate Authority which actually does the work of signing user/server certs
+  # Good practice as a comprimised intermediate CA can be invalidated
+  # ---
+  export CA=${iCA}
+  export CA_POLICY=policy_loose
+  printf "\n\n---- Generating Intermediate CA ($iCA)  ----\n\n"
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh]: fixed

File: open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh [hunk 5]
```diff
@@ -43,112 +43,114 @@ CA_CHAIN=EgeriaCAChain
...
 
 # ---
 # Server Certs
 # ---
-for cert in ${SERVERCERTS}
-do
+for cert in ${SERVERCERTS}; do
+
   FNAME=Egeria${cert}
+  rm -f ${FNAME}*
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh]: will not fix - we need globbing

File: open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh [hunk 6]
```diff
@@ -43,112 +43,114 @@ CA_CHAIN=EgeriaCAChain
...
-# Deploy this anywhere that needs to 'trust' the certs we create later. It should be sufficient
-printf "\n\n---- Create truststore (rootCA) ----\n\n"
-
-# -- will use keytool instead, can't seem to get a cert exported via openssl ...
-#openssl pkcs12 -export -in ${rootCA}/certs/${rootCA}.cert.pem -out ${rootCA}.p12 \
-#          -passin pass:${KEYPASS} -passout pass:${KEYPASS} -nokeys -cacerts
-keytool -importcert -noprompt -keystore ${rootCA}.p12 -storepass ${STOREPASS} -alias ${rootCA} \
-        -file ${rootCA}/certs/${rootCA}.cert.pem
-
+if [ ! -z $CA_CLEAN ]; then
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/certificates/gensamplecerts.sh:51]: will not fix, we need globbing

**Pull request merged**